pulsar-broker/src/test/java/org/apache/pulsar/broker/TopicEventsListenerTest.java

```diff
@@ -0,0 +1,672 @@
+        String topicName = "persistent://" + namespace + "/P-NP";
+        topicNameToWatch = topicName + "-partition-1";
+
+        admin.topics().createPartitionedTopic(topicName, 2);
+        triggerPartitionsCreation(topicName);
+
+        Awaitility.waitAtMost(10, TimeUnit.SECONDS).untilAsserted(() ->
+                Assert.assertEquals(events.toArray(), new String[]{
+                        "LOAD__BEFORE",
+                        "CREATE__BEFORE",
+                        "CREATE__SUCCESS",
+                        "LOAD__SUCCESS"
+                })
+        );
+
+        events.clear();
+        admin.topics().deletePartitionedTopic(topicName, true);
+
+        Awaitility.waitAtMost(10, TimeUnit.SECONDS).untilAsserted(() ->
+                Assert.assertEquals(events.toArray(), new String[]{
+                        "DELETE__BEFORE",
+                        "UNLOAD__BEFORE",
+                        "UNLOAD__SUCCESS",
+                        "DELETE__SUCCESS"
+                })
+        );
+    }
+
+    @Test
+    public void testEventsPersistentPartitionedTopicWithUnloadForce() throws Exception {
+        String topicName = "persistent://" + namespace + "/P-NP";
+        topicNameToWatch = topicName + "-partition-0";
+
+        admin.topics().createPartitionedTopic(topicName, 2);
+        triggerPartitionsCreation(topicName);
+
+        Awaitility.waitAtMost(10, TimeUnit.SECONDS).untilAsserted(() ->
+                Assert.assertEquals(events.toArray(), new String[]{
+                        "LOAD__BEFORE",
+                        "CREATE__BEFORE",
+                        "CREATE__SUCCESS",
+                        "LOAD__SUCCESS"
+                })
+        );
+
+        events.clear();
+        admin.topics().unload(topicName);
+
+        Awaitility.waitAtMost(10, TimeUnit.SECONDS).untilAsserted(() ->
+                Assert.assertEquals(events.toArray(), new String[]{
+                        "UNLOAD__BEFORE",
+                        "UNLOAD__SUCCESS"
+                })
+        );
+
+        events.clear();
+        admin.topics().deletePartitionedTopic(topicName, true);
+
+        Awaitility.waitAtMost(10, TimeUnit.SECONDS).untilAsserted(() ->
+                Assert.assertEquals(events.toArray(), new String[]{
+                        "DELETE__BEFORE",
+                        "DELETE__SUCCESS"
+                })
+        );
+    }
+
+    @Test
+    public void testNonPartitionedTopicAutoGC() throws Exception {
+        topicNameToWatch = "persistent://" + namespace + "/P-NP";
+        admin.topics().createNonPartitionedTopic(topicNameToWatch);
+        admin.namespaces().setInactiveTopicPolicies(namespace,
+                new InactiveTopicPolicies(InactiveTopicDeleteMode.delete_when_no_subscriptions, 1, true));
+
+        // Remove retention
+        admin.namespaces().setRetention(namespace, new RetentionPolicies());
+        events.clear();
+
+        Thread.sleep(1500);
```

Comment:
Why did you sleep here? (The same question goes for `testPartitionedTopicAutoGC`)

Reply:
retention is not propagated/set immediately after  admin.namespaces().setRetention returns.
I'll look for a way to check for it w/awaitility